caffe2/operators/shape_op.h

```diff
@@ -19,18 +19,18 @@ class ShapeOp : public Operator<Context> {
 
   bool RunOnDevice() override {
     auto& data = Input(DATA);
-    auto* output = Output(0);
+
     int numDims = data.dim();
     int numAxes = axes_.size();
     if (numAxes == 0) {
-      output->Resize(numDims);
+      auto* output = Output(0, {numDims}, at::dtype<int64_t>());
       int64_t* output_data = output->template mutable_data<int64_t>();
       context_.CopyBytesSameDevice(
           numDims * sizeof(int64_t), data.sizes().data(), output_data);
       return true;
     }
 
-    output->Resize(numAxes);
+    auto* output = Output(0, {numAxes}, at::dtype<int64_t>());
```

Comment:
I'm confused by this code. If `numAxes == 0`, then it first resizes it to `numDims`, adds some data, and then directly resizes it to `numAxes`, which is zero? Why write some data to it before resizing to zero? Should this code down here maybe live in an `else` branch of the above condition?

Reply:
What do you mean? there was a `return true` in the above if statement, I guess you missed that?